base/file.jl

```diff
@@ -197,10 +197,11 @@ tempdir() = dirname(tempname())
 
 # Create and return the name of a temporary file along with an IOStream
 function mktemp(parent=tempdir())
-    b = joinpath(parent, "tmpXXXXXX")
-    p = ccall(:mkstemp, Int32, (Cstring,), b) # modifies b
+    b = joinpath(parent, "tmpXXXXXX").data
+    0x00 in b && throw(ArgumentError("path cannot contain NUL"))
+    p = ccall(:mkstemp, Int32, (Ptr{UInt8},), b) # modifies b
```

Comment:
It was a problem for one of the other changes. I can leave this change out of this PR though.